### PR TITLE
feat: get preferential gauge for given pool id

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
still needs unit tests

### CODE REVIEW

File: tests/test_pools_gauges.py [hunk 1]
```diff
@@ -4,8 +4,17 @@
 
 EXAMPLE_PREFERENTIAL_GAUGES = {
```
Comment (Tritium-VLK) [inline tests/test_pools_gauges.py:5]: How did you pick these examples?  Did you find edge cases/problem children?  I think if we asked people to think about it we could come up with a few.

**Review by Tritium-VLK — COMMENTED** [review]
Code looks good.  Tests would be better if they tested a bunch of edge cases and included some comments to help describe what those edge cases are.

one of the ways they are changed is via the eventadder.  I think you could look there to find cases where we had to fix something.   Ping me if you need help finding it.

**Review by Tritium-VLK — APPROVED** [review]
